diff --git a/examples/helloworld-myplat/Cargo.toml b/examples/helloworld-myplat/Cargo.toml
index 0117eef..cc3a753 100644
--- a/examples/helloworld-myplat/Cargo.toml
+++ b/examples/helloworld-myplat/Cargo.toml
@@ -9,12 +9,13 @@ aarch64-bsta1000b = ["dep:axplat-aarch64-bsta1000b"]
 aarch64-phytium-pi = ["dep:axplat-aarch64-phytium-pi"]
 aarch64-qemu-virt = ["dep:axplat-aarch64-qemu-virt"]
 aarch64-raspi4 = ["dep:axplat-aarch64-raspi"]
+aarch64-dyn = ["dep:axplat-aarch64-dyn"]
 loongarch64-qemu-virt = ["dep:axplat-loongarch64-qemu-virt"]
 riscv64-qemu-virt = ["dep:axplat-riscv64-qemu-virt"]
 x86-pc = ["dep:axplat-x86-pc"]
 
 [dependencies]
-axstd = {workspace = true, features = ["myplat"], optional = true}
+axstd = { workspace = true, features = ["alloc", "myplat"], optional = true }
 cfg-if = "1.0"
 
 [target.'cfg(target_arch = "x86_64")'.dependencies]
@@ -24,6 +25,7 @@ axplat-x86-pc = {workspace = true, features = ["smp", "irq"], optional = true}
 axplat-aarch64-bsta1000b = {workspace = true, features = ["smp", "irq"], optional = true}
 axplat-aarch64-phytium-pi = {workspace = true, features = ["smp", "irq"], optional = true}
 axplat-aarch64-qemu-virt = {workspace = true, features = ["smp", "irq"], optional = true}
+axplat-aarch64-dyn = { git = "https://github.com/arceos-hypervisor/axplat-aarch64-dyn.git", branch = "v03", features = ["smp", "irq"], optional = true }
 axplat-aarch64-raspi = {workspace = true, features = ["smp", "irq"], optional = true}
 
 [target.'cfg(target_arch = "riscv64")'.dependencies]
diff --git a/examples/helloworld-myplat/src/main.rs b/examples/helloworld-myplat/src/main.rs
index ff1198b..15887ec 100644
--- a/examples/helloworld-myplat/src/main.rs
+++ b/examples/helloworld-myplat/src/main.rs
@@ -16,6 +16,8 @@ cfg_if::cfg_if! {
         extern crate axplat_riscv64_qemu_virt;
     } else if #[cfg(all(target_arch = "loongarch64", feature = "loongarch64-qemu-virt"))] {
         extern crate axplat_loongarch64_qemu_virt;
+    } else if #[cfg(all(target_arch = "aarch64", feature = "aarch64-dyn"))] {
+        extern crate axplat_aarch64_dyn;
     } else {
         #[cfg(target_os = "none")] // ignore in rust-analyzer & cargo test
         compile_error!("No platform crate linked!\n\nPlease add `extern crate <platform>` in your code.");
diff --git a/modules/axhal/Cargo.toml b/modules/axhal/Cargo.toml
index 9b2f312..e2538ea 100644
--- a/modules/axhal/Cargo.toml
+++ b/modules/axhal/Cargo.toml
@@ -41,7 +41,7 @@ ipi = ["irq"]
 [dependencies]
 axalloc = {workspace = true, optional = true}
 axconfig = {workspace = true}
-axcpu = "0.2"
+axcpu = { git = "https://github.com/arceos-org/axcpu.git", tag="dev-v03" }
 axlog = {workspace = true}
 axplat.workspace = true
 cfg-if = "1.0"
diff --git a/scripts/make/features.mk b/scripts/make/features.mk
index 7a7f153..9a01ebb 100644
--- a/scripts/make/features.mk
+++ b/scripts/make/features.mk
@@ -54,7 +54,7 @@ ifeq ($(BUS),mmio)
   ax_feat += bus-mmio
 endif
 
-ifeq ($(shell test $(SMP) -gt 1; echo $$?),0)
+ifeq ($(shell test $(SMP) -gt 0; echo $$?),0)
   lib_feat += smp
 endif
 
